src/shared/copy.c

```diff
@@ -765,6 +765,32 @@ static int memorize_hardlink(
         return 1;
 }
 
+static int prepare_nocow(int fdf, const char *from, int fdt, unsigned *chattr_mask, unsigned *chattr_flags) {
+        unsigned attrs = 0;
+        int r;
+
+        assert(fdf >= 0 || fdf == AT_FDCWD);
+        assert(fdt >= 0);
+        assert(!!chattr_mask == !!chattr_flags);
+
+        if (chattr_mask && FLAGS_SET(*chattr_mask, FS_NOCOW_FL))
+                return 0;
+
+        r = read_attr_at(fdf, from, &attrs);
+        if (r < 0 && !ERRNO_IS_NOT_SUPPORTED(r) && r != -ELOOP)
```

Comment:
hmm, ELOOP?

Reply:
This happens when the source is a symlink

Reply:
please add comment explaining that